windows-gaming-driver/src/driver/controller.rs

```diff
@@ -32,13 +35,23 @@ enum State {
     Resuming,
 }
 
+#[derive(PartialEq, Eq, Clone, Copy)]
+enum IoState {
+    Detached,
+    LightEntry,
+    LightFallback,
```

Comment:
Consider renaming this to something `LightAwaitingUpgrade` to better describe what it does.